proxy_telnet.go

```diff
@@ -0,0 +1,127 @@
+package glutton
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/reiver/go-telnet"
+	log "go.uber.org/zap"
+)
+
+//telnetProxy Struct
+type telnetProxy struct {
+	logger      *log.Logger
+	curConn     net.Conn
+	proxyclient *telnet.Client
+	hostconn    *telnet.Conn
+	glutton     *Glutton
+	host        string
+}
+
+//NewTelnetProxy Create a new Telnet Proxy Session
+func (g *Glutton) NewTelnetProxy(destinationURL string) (err error) {
+
+	t := &telnetProxy{
+		logger:  g.logger,
+		glutton: g,
+	}
+
+	dest, err := url.Parse(destinationURL)
+	if err != nil {
+		t.logger.Error("[telnet.prxy] failed to parse destination address, check config.yaml")
+		fmt.Println(fmt.Sprintf("[telnet.prxy ]   failed to parse destination address, check config.yaml"))
+		return err
+	}
+	t.logger.Info("[telnet proxy] " + dest.Host)
+	t.host = dest.Host
+	g.telnetProxy = t
+	return
+}
+
+//handle Telnet Proxy handler
+func (t *telnetProxy) handle(ctx context.Context, conn net.Conn) (err error) {
+	ended := false
+	g := t.glutton
+	tcpAddr, err := net.ResolveTCPAddr("tcp", t.host)
+	if err != nil {
+		println("ResolveTCPAddr failed:", err.Error())
+		return
+	}
+	hconn, err := net.DialTCP("tcp", nil, tcpAddr)
+	if err != nil {
+		t.logger.Error(fmt.Sprintf("[telnet proxy  ]  Connection error: %v", err))
+		fmt.Println(fmt.Sprintf("[telnet proxy ]  Coonection error: %v", err))
+		return
+	}
+	n := 0
+	go func() {
+		defer func() {
+			ended = true
+			conn.Close()
+			hconn.Close()
+		}()
+		for {
+			if ended == true || hconn == nil || conn == nil {
+				break
+			}
+			reply := make([]byte, 8*1024)
```

Comment:
what if more is send?

Reply:
Please notice line no. 59 in which I have called this function as a go routine. That means that this function is running asynchronously. I will continue to fetch the data from connection and continuously output to screen. So it it does not matter whatever value I put here or whatever the size of array is.

I hope I have made myself clear.